### PR TITLE
Fix ScummVM in the menu to match the DAT

### PR DESCRIPTION
**Pull request by RobLoach**
The game's description is usually more human-friendly than the given rom name. You can see  this happen in https://github.com/libretro/libretro-database/pull/221 . Doesn't affect any of the other DATs we have, as the key is mirrored.

scummvm
scummvm2

... With this change, it ends up being correct....

scummvm3

### CODE REVIEW

**Comment by t-paul** [comment]
Wouldn't it be better to fix the data file instead?

Having a description that can't be used as a longer text to describe the game seems strange:
https://github.com/libretro/libretro-database/blob/master/dat/Sinclair%20-%20ZX%2081.dat#L6077

**Comment by RobLoach** [comment]
Thanks for pointing out Sinclair, didn't see any that didn't mirror the descriptions. Updating the DAT would be difficult, as it's quite large. Perhaps use description if it is <=90 characters?

**Comment by Kivutar** [comment]
We can't do this, the common consoles set have very long descriptions.

File: tasks/task_database.c [hunk 1]
```diff
@@ -313,11 +313,10 @@ static int database_info_list_iterate_found_match(
...
    if(!playlist_entry_exists(playlist, entry_path_str, db_crc))
    {
-      // For the playlist item name, use the game's
-      // description if it's less then 60 characters
-      // long. Otherwise, just use the name.
+      // For ScummVM, use the game description, for every
+      // other system, use the DAT's game name.
       char* name;
-      if (strlen(db_info_entry->description) <= 60)
+      if (strcmp(db_playlist_base_str, "ScummVM.lpl") == 0)
```
Comment (RobLoach) [inline tasks/task_database.c]: > use another field to store long descriptions.

Now it only uses the description field for ScummVM. Having the DAT contain the name means you can run clrmamepro on the DAT on your game datafiles, and it structures the files in the folders for ScummVM correectly.

**Comment by Kivutar** [comment]
Ok I understand. So I was certainly wrong when I choose the description field to store a long description in the metadat.

**Comment by RobLoach** [comment]
> I was certainly wrong when I choose the description field to store a long description in the metadat.

You were certainly not wrong, since there is no official spec on how to correctly use DAT files. They're not consistent across different DATs. No-Intro mirrors the name/description, ScummVM uses name as a the Game ID and the description as the Human-readable name, Sinclair uses the name as a name and the description as a description (who knew!).

All we have to do here is just be specific about how its used in the DAT, and adjust accordingly. Thanks for the reviews, you two.

**Comment by lollo78** [comment]
hi @RobLoach!
Lovely idea for ScummVM.
I downloaded and compiled the latest stuff (Retroarch, libretro database...) and I tried you solution with one of my games.
I renamed the game folder using the ScummVM ID (tentacle-it) and RA scanned it correctly, but the game name into the playlist is always the ID (tentacle-it). 
How can I show the game description instead of the ID?
Thx a lot

**Comment by RobLoach** [comment]
> How can I show the game description instead of the ID?

This does it: https://github.com/libretro/RetroArch/pull/3215

**Comment by lollo78** [comment]
I'm just a little confused :D 
This mod is not merged?
If yes, why I continue to see the games ID like the games name into the ScummVM.pl?

**Comment by RobLoach** [comment]
The PR is not merged, no. Would appreciate your thoughts on it in there, though. May help it get in.

**Comment by lollo78** [comment]
OK for testing.
Please let me know what I have to compile or download to do it.
Thx

**Comment by RobLoach** [comment]
> let me know what I have to compile or download to do it.

1. Remove ~/.config/retroarch/playlists/ScummVM.lpl
2. Compile RetroArch...

```
git clone https://github.com/RobLoach/RetroArch.git RetroArch-RobLoach
cd RetroArch-RobLoach
git checkout game-description
make
./retroarch
...

3. Scan your `tentacle` directory, and Day of the Tentacle should show up in the menu.

**Comment by lollo78** [comment]
Ah OK, I have to compile RA from your repository.
I'll do it as soon as possible. I'm away from home now.
Thx

**Comment by lollo78** [comment]
@RobLoach 
Hi dear, for me the best way to test your ScummVM fix is to fork official RA and add a commit with your fix. So I can compile it with my actual build enviroinment.
I done this 
https://github.com/lollo78/RetroArch/commit/5f77bf5c13ab3e939e865fce7391915ba5b5b8de
but after scanning tentacle-it I see always tentacle-it like game name, no description.
What I have missed of your code?
Please tell me what I have to change again in my RA fork to test your fix.
Thx a lot

**Comment by RobLoach** [comment]
Missing this change:
https://github.com/lollo78/RetroArch/commit/5f77bf5c13ab3e939e865fce7391915ba5b5b8de#commitcomment-18464946

**Comment by RobLoach** [comment]
We could likely fix this in the DAT. Would need a script to clean it up, though. Going to close this for now.

**Pull request closed without merging**